### PR TITLE
Fix local participant data not updated after forced reconnection

### PR DESCRIPTION
**Pull request by danxuliu**
After a forced reconnection both the Nextcloud and the signaling session ID change. Neither the `peerId` in LocalCallParticipantModel nor the session in the actor store were updated, which caused some issues.

It was mostly visible in the speaking time of the local participant after a forced reconnection, as [the speaking time was no longer updated](https://github.com/nextcloud/spreed/pull/10068#pullrequestreview-1565565757) (due to being set to the previous Nextcloud session ID, which is no longer in the call or conversation); something similar happens for the raised hand. There is also a subtler issue with the typing notifications (being sent to both the previous and current session of the self participant, [as they do not match](https://github.com/nextcloud/spreed/blob/f755a2c8edbdd89581912f4c4d35648a1e126d81/src/utils/SignalingTypingHandler.js#L93-L98)).

I am not fond of modifying the store directly from the signaling; probably a better approach would be to add a `rejoinConversation` or `forceReconnect` action to the store (to be called instead of `signaling.forceReconnect`) which would call the service and this, in turn, call the API and the signaling, and then update the store from the returned response (that is, the same done for joining a conversation). But refactoring all that is far from an easy task so, for now... modifying the store from the signaling should suffice :see_no_evil:

## How to test (common setup)
- Add a function to force reconnect from the console in [_src/utils/webrtc/webrtc.js_](https://github.com/nextcloud/spreed/blob/e1b60c0e360c322db078a0b8d6f349382338c61d/src/utils/webrtc/webrtc.js#L965):
```
	webrtc.forceReconnect = function() {
		forceReconnect(signaling)
	}
```
- Rebuild the code
- Setup the HPB

## How to test (scenario 1)

- Start a call
- Enable audio and start speaking
- In the browser console, run `OCA.Talk.SimpleWebRTC.forceReconnect()`

### Result with this pull request

The speaking time for the local participant is updated

### Result without this pull request

The speaking time for the local participant is neither updated nor even shown

## How to test (scenario 2)

- Start a call
- In the browser console, run `OCA.Talk.SimpleWebRTC.forceReconnect()`
- Raise the hand

### Result with this pull request

The raised hand is shown in the sidebar

### Result without this pull request

The raised hand is not shown in the sidebar

### CODE REVIEW

**Comment by Antreesy** [comment]
Test | Master | This PR | Common issues/ follow-ups
--|--|--|--
Scenario 1 (Speaking time) | new session id for local participant is missing  | :green_circle:  new session id for local participant is added | previously elapsed time is still in the store, but not updated and not visible anymore, though it's the same participant
Scenario 2 (Raised hand) | after reconnection raised hand is no longer shown locally | :green_circle: works fine | when reconnected, raised hand is dropped, but local web-interface still shows it  
Scenario 3 (Typing indicator) | new session id for local participant is missing  | :green_circle: new session id for local participant is added | nothing broken, bc we don't follow local participant

File: src/components/CallView/CallView.vue [hunk 1]
```diff
@@ -362,6 +362,13 @@ export default {
 		},
 	},
 	watch: {
+		'localCallParticipantModel.attributes.peerId'(newValue, previousValue) {
+			const index = this.screens.indexOf(previousValue)
+			if (index !== -1) {
+				this.screens.splice(index, 1, newValue)
```
Comment (Antreesy) [inline src/components/CallView/CallView.vue:368]:
```suggestion
				this.screens[index] = newValue
```

Same result, but with direct approach will be better, i suppose

**Review by Antreesy — APPROVED** [review]
Nothing blocking from my POV, everything works better now)

But I'd like to work on a couple of things raised in the previos comment - maybe we could store speaking status and elapsed time by attendeeId, not the sessionId? That way we'll keep elapsed already time, but there might be some issues, if participant connects and speaks from different devices (rare case, but still possible?)

File: src/components/CallView/CallView.vue (hunk 1 above)
Comment (danxuliu) [reply src/components/CallView/CallView.vue]: :facepalm: Yes, totally agree, I do not know what I was thinking on xD

**Comment by danxuliu** [comment]
> Scenario 1 (Speaking time) - previously elapsed time is still in the store, but not updated and not visible anymore, though it's the same participant

I thought about adding the previously elapsed time to the new session. But as it was only a local fix and the speaking time would be still reset from the point of view of other participants I ended not doing that :shrug:

> Scenario 2 (Raised hand) - when reconnected, raised hand is dropped, but local web-interface still shows it

Probably the fix would be just sending the hand raised state again (if the hand is raised) after a reconnection. I will open an issue about that.

> But I'd like to work on a couple of things raised in the previos comment - maybe we could store speaking status and elapsed time by attendeeId, not the sessionId? That way we'll keep elapsed already time, but there might be some issues, if participant connects and speaks from different devices (rare case, but still possible?)

That is a good idea. In fact maybe the speaking time of all the different devices should be added to show a single speaking time for the attendee :thinking: ; except in rare cases in which the same user is two different persons from separate devices I guess that a unified speaking time makes sense (also a single entry is shown in the participant list when the same user is connecting from different devices at the same time).

The speaking time will still be reset anyway for guests after a forced reconnection, but that is how things are (at least until we can identify guests).

Nevertheless, I would suggest to change that in a follow up pull request :-)

**Pull request merged**